src/main/java/com/github/fge/jsonpatch/JsonPatch.java

```diff
@@ -89,12 +89,36 @@
  * <p><b>IMPORTANT NOTE:</b> the JSON Patch is supposed to be VALID when the
  * constructor for this class ({@link JsonPatch#fromJson(JsonNode)} is used.</p>
  */
-public final class JsonPatch
+public class JsonPatch
     implements JsonSerializable
 {
-    private static final MessageBundle BUNDLE
+    protected static final MessageBundle BUNDLE
         = MessageBundles.getBundle(JsonPatchMessages.class);
 
+    private static final ObjectMapper mapper;
+
+    static {
+        final ObjectMapper newMapper = JacksonUtils.newMapper();
+        newMapper.registerSubtypes(
+            new NamedType(AddOperation.class, AddOperation.OPERATION_NAME),
+            new NamedType(CopyOperation.class, CopyOperation.OPERATION_NAME),
+            new NamedType(MoveOperation.class, MoveOperation.OPERATION_NAME),
+            new NamedType(RemoveOperation.class, RemoveOperation.OPERATION_NAME),
+            new NamedType(ReplaceOperation.class, ReplaceOperation.OPERATION_NAME),
+            new NamedType(TestOperation.class, TestOperation.OPERATION_NAME)
+        );
+        mapper = newMapper;
+    }
+
+    public static ObjectReader getReader()
```

Comment:
We need to get the reader if we want to deserialize using Jackson not using fromJson (for example, if you want to deserialize an object that takes JsonPatch in its constructor)